### DISCUSSION TITLE
Why does clangd always choose its built-in headers for some of the headers?

### DISCUSSION THREAD

**Discussion by palapapa** [clangd/clangd#1969] [Q&A] [answered]
I have found that for some headers like `stddef.h`, `stdint.h`, and `stdbool.h`, clangd will always choose the ones that are in its installation directory. For example, the path of `stddef.h` is shown as `c:\Users\User\AppData\Roaming\Code\User\globalStorage\llvm-vs-code-extensions.vscode-clangd\install\15.0.3\clangd_15.0.3\lib\clang\15.0.3\include\stddef.h`, while for `stdio.h` it's `C:\mingw64\x86_64-w64-mingw32\include\stdio.h`, which is correct. [Here](https://clangd.llvm.org/guides/system-headers#what-are-system-headers-) it does mention that clangd ships with its own built-in headers, so I don't know if this is the expected behavior.

My compile flags(specified through the `fallbackFlags` option in VSCode):
```json
"clangd.fallbackFlags": [
    "-Weverything",
    "-Wno-used-but-marked-unused",
    "-Wno-declaration-after-statement",
    "-Wno-vla",
    "-Wno-missing-prototypes",
    "-Wno-c++98-compat",
    "-Wno-reserved-macro-identifier",
    "-Wno-padded",
    "-Wno-unsafe-buffer-usage",
    "-Wno-empty-translation-unit",
    "-target",
    "x86_64-w64-windows-gnu",
    "-xc",
    "-std=c17"
  ]
```

**Comment by HighCommander4** [comment, marked as answer]
Yes, this is expected. Headers in the built-in header directory are closely coupled to the compiler front-end, such that clang's built-in headers need to be used when parsing a file with the clang front-end.
